### PR TITLE
[Fix] Fix inconsistent float precision between mmdet and mmcv

### LINKED ISSUE #9569: [Bug] Inconsistent float precision in training log
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection

### Environment

sys.platform: linux
Python: 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31) [GCC 7.3.0]
CUDA available: True
GPU 0,1,2,3,4,5,6,7: Tesla V100-SXM2-32GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.2, V10.2.8
GCC: gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
PyTorch: 1.7.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

TorchVision: 0.8.0
OpenCV: 4.5.3
MMCV: 1.5.0
MMCV Compiler: GCC 7.4
MMCV CUDA Compiler: 10.2
MMDetection: 2.21.0+

### Reproduces the problem - code sample

training on any dataset

### Reproduces the problem - command or script

training on any dataset

### Reproduces the problem - error message

2022-12-28 11:15:09,949 - mmdet - INFO -
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.00041
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = **0.00150**
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.00013
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.00000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.00009
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.00101
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.09577
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.09577
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.09577
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.00000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.05205
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.17818

2022-12-28 11:15:10,003 - mmdet - INFO - Exp name: autoassign_r50_fpn_8x2_1x_mae_x-bkpd.py
2022-12-28 11:15:10,003 - mmdet - INFO - Epoch(val) [100][1177] bbox_mAP: 0.0000, bbox_mAP_50: **0.0010**, bbox_mAP_75: 0.0000, bbox_mAP_s: 0.0000, bbox_mAP_m: 0.0000, bbox_mAP_l: 0.0010, bbox_mAP_copypaste: 0.000 0.001 0.000 0.000 0.000 0.001

-------------------------------

I am training on my custom dataset and I notice that the mAPs in log have different float precision, which results in different scores after rounding off.
For example, mAP_50 is 0.00150 before and 0.0010 after.
This is much more severe when the performance of the model is bad. A minor difference in float precision will have more significant difference when selecting the best model.

### Additional information

图片

### PR DESCRIPTION
**Pull request by jamiechoi1995**
In coco.py and lvis.py, the float precision of bbox mAP of eval_result of evaluate() function is .3f,
However, in the float precision when printing log in mmcv is .4f, 
see:
https://github.com/open-mmlab/mmcv/blob/ea173c9f07f0abf6873d2b7d786fb6411843cf00/mmcv/runner/hooks/logger/text.py#L178-L179

This PR suggests using the same .4f float precision. Another solution is not to limit the float precision in eval_result.

Fix #9569

### CODE REVIEW

File: mmdet/datasets/coco.py [hunk 1]
```diff
@@ -525,7 +525,7 @@ def evaluate_det_segm(self,
 
                 for item in metric_items:
                     val = float(
-                        f'{cocoEval.stats[coco_metric_names[item]]:.3f}')
+                        f'{cocoEval.stats[coco_metric_names[item]]:.4f}')
```
Comment (RangiLyu) [inline mmdet/datasets/coco.py:528]: Since you changed the float precision of the metric, the unit test should also be updated.

**Review by RangiLyu — APPROVED** [review]
